tests/client/notification_service_tests.cpp

```diff
@@ -9,32 +9,81 @@
 #include <string>
 
 #include <boost/asio.hpp>
+#include <boost/bind/bind.hpp>
 #include <chrono>
 #include <boost/date_time/posix_time/posix_time.hpp>
 
 namespace xpx_chain_sdk::tests {
 
 #define TEST_CLASS NotificationService
 
-    boost::asio::io_context io;
-    boost::asio::deadline_timer timer(io, boost::posix_time::seconds(60));
-
     ClientData clientData;
     auto client = xpx_chain_sdk::getClient(std::make_shared<xpx_chain_sdk::Config>(getTestConfiguration()));
     auto account = getTestAccount(clientData.privateKey);
 
+    void wait_Block(const boost::system::error_code&, int* count, bool* isReceived, boost::asio::deadline_timer* timer){
```

Comment:
Better to use references instead of raw pointers = )

Reply:
Actually, I tried to use references before deciding to use pointers. However, it shows this error. I couldn't solve it before. Do you know what is wrong with the following code ?

![image](https://user-images.githubusercontent.com/86714622/135982491-6cbd6327-7f63-40f6-a6f1-943a8cb291ae.png)